Fuser/runner.py

```diff
@@ -229,6 +194,84 @@ def run_candidate(
         except Exception:
             pass
 
+    return rc, t_finished
+
+
+def run_candidate(
+    artifacts_code_path: Path,
+    run_root: Path,
+    timeout_s: int,
+    isolated: bool,
+    deny_network: bool,
+    cancel_event: Optional["threading.Event"] = None,
+) -> RunResult:
+    """
+    Execute a candidate program in a fresh run directory under run_root.
+    - Copies artifacts_code_path to run_dir/candidate_main.py
+    - Runs [sys.executable, '-u', 'candidate_main.py'] with optional -I (isolated)
+    - If deny_network, injects sitecustomize.py to block sockets and do NOT use -I
+    - Captures stdout/stderr to files; kills on timeout or cancel_event
+    - Classifies pass/fail according to design precedence
+    """
+    run_dir = (
+        run_root
+        / f"attempt_{int(time.time() * 1000)}_{os.getpid()}_{random.randint(0, 9999):04d}"
+    )
+    run_dir.mkdir(parents=True, exist_ok=False)
+
+    # Prepare working files. We intentionally avoid the name "code.py" here because
+    # Python's stdlib exposes a module with that name, and PyTorch's import stack
+    # (via pdb -> code) would accidentally load the candidate file instead of the
+    # stdlib module, leading to partially initialised torch packages.
+    exec_filename = "candidate_main.py"
+    code_dst = run_dir / exec_filename
+    st = artifacts_code_path.lstat()
+    if not stat.S_ISREG(st.st_mode) or stat.S_ISLNK(st.st_mode):
+        raise ValueError("artifacts_code_path must be a regular file (no symlink)")
+    shutil.copy2(artifacts_code_path, code_dst)
+
+    if deny_network:
+        _write_sitecustomize_block_network(run_dir)
```

Comment:
`deny_network` is likely not enforced in multiprocessing mode.
In the child process (runner_util), before executing candidate code:
If (run_dir / "sitecustomize.py") exists, explicitly import/execute it (or otherwise apply the same socket-blocking logic) before candidate code runs.
Keep the “no -I when deny_network” behavior as-is (that part is fine).